main.go

```diff
@@ -56,92 +65,35 @@ func loadConfig() (string, string, string, bool) {
 	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
 	viper.AutomaticEnv()
 
-	var tlsSkipVerify bool
-	tlsSkipVerify = viper.GetBool("tls.skip_verify")
-
-	host, user, token := viper.GetString("core.host"), viper.GetString("core.user"), viper.GetString("core.token")
-	if host != "" && user != "" && token != "" {
-		return host, user, token, tlsSkipVerify
-	} else if host != "" && token != "" {
-		user = getUser(host, token, tlsSkipVerify)
-		return host, user, token, tlsSkipVerify
-	}
-
-	// Attempt to auto-configure for GitLab CI
-	host, user, token = config.CI()
-	if host != "" && user != "" && token != "" {
-		return host, user, token, tlsSkipVerify
-	}
-
 	if _, ok := viper.ReadInConfig().(viper.ConfigFileNotFoundError); ok {
-		if err := config.New(path.Join(confpath, "lab.toml"), os.Stdin); err != nil {
+		err := config.New(path.Join(confpath, "lab.toml"), os.Stdin)
+		if err != nil {
 			log.Fatal(err)
 		}
 
-		if err := viper.ReadInConfig(); err != nil {
+		err = viper.ReadInConfig()
+		if err != nil {
 			log.Fatal(err)
 		}
 	}
 
-	c := viper.AllSettings()
-	var cfg map[string]interface{}
-	switch v := c["core"].(type) {
-	// Most run this is the type
-	case []map[string]interface{}:
-		cfg = v[0]
-	// On the first run when the cfg is created it comes in as this type
-	// for whatever reason
-	case map[string]interface{}:
-		cfg = v
-	}
+	var tlsSkipVerify bool
+	tlsSkipVerify = viper.GetBool("tls.skip_verify")
 
-	for _, v := range []string{"host", "token"} {
-		if cv, ok := cfg[v]; !ok {
-			log.Println(cv)
-			log.Fatalf("missing config value core.%s in %s", v, viper.ConfigFileUsed())
-		}
-	}
+	host = viper.GetString("core.host")
+	user = viper.GetString("core.user")
+	token = viper.GetString("core.token")
 
-	var tls map[string]interface{}
-	switch v := c["tls"].(type) {
-	// Most run this is the type
-	case []map[string]interface{}:
-		tls = v[0]
-	// On the first run when the cfg is created it comes in as this type
-	// for whatever reason
-	case map[string]interface{}:
-		tls = v
-	}
-	if v, ok := tls["skip_verify"]; ok {
-		tlsSkipVerify = v.(bool)
+	if host != "" && user != "" && token != "" {
+		return host, user, token, tlsSkipVerify
 	}
 
-	// Set environment overrides
-	// Note: the code below that uses `cfg["host"]` to access these values
-	// is tough to simplify since cfg["host"] is accessing the array "core"
-	// and viper.GetString("core.host") is expecting a non-array so it
-	// doens't match
-	if v := viper.GetString("core.host"); v != "" {
-		cfg["host"] = v
-	}
-	if v := viper.GetString("core.token"); v != "" {
-		cfg["token"] = v
+	user = getUser(host, token, tlsSkipVerify)
+	if os.Getenv("LAB_CORE_TOKEN") != "" || os.Getenv("LAB_CORE_HOST") != "" {
```

Comment:
I'm a bit confused by this part: why core.user is being set only when LAB_CORE_TOKEN or LAB_CORE_HOST are set?

Reply:
I'm just writing the core.user value because it's new.  I'll double check that this works as expected and writes the whole config.

Reply:
Same thing as the previous comment.  I've fixed this so that it works when LAB_CORE_TOKEN or LAB_CORE_HOST is set.  I've verified that the entire config is rewritten.